dataprofiler/profilers/profile_builder.py

```diff
@@ -2112,6 +2130,171 @@ def _update_chi2(self):
 
         return chi2_mat
 
+    def _update_null_replication_metrics(self, clean_samples):
+        """
+        Calculate metrics needed for replicating null values in capitalone/synthetic-data.
+
+        Required for running LDA based binary classifier where predicted class label indicates whether
+        a value of a column should be NaN (1) or not (0).
+
+        :param clean_samples: input cleaned dataset
+        :type clean_samples: dict
+        """
+        data = pd.DataFrame(clean_samples).apply(pd.to_numeric, errors="coerce")
+
+        get_data_type = lambda profile: profile.profiles[
+            "data_type_profile"
+        ].selected_data_type
+        get_data_type_profiler = lambda profile: profile.profiles[
+            "data_type_profile"
+        ]._profiles[get_data_type(profile)]
+
+        total_row_sum = np.asarray(
+            [get_data_type_profiler(profile).sum for profile in self._profile]
+        )
+
+        for col_id, profile in enumerate(self._profile):
+            null_count = getattr(profile, "null_count")
+            if null_count == 0:
+                # No missing values to replicate
+                continue
+
+            sample_size = getattr(profile, "sample_size")
+            true_count = sample_size - null_count
+
+            # null_count and sample_size get updated on profile_update
+            # Therefore the priors are calculated from the updated values
+            prior_null = null_count / sample_size
+            prior_not_null = true_count / sample_size
+
+            # Gets list of null indices of the entire dataset
+            null_type_dict = getattr(profile, "null_types_index")
+            null_indices = set.union(*null_type_dict.values())
+            null_indices = np.asarray(list(null_indices))
+
+            # Keep only the null indices inside the chunk (reverse index shift)
+            if profile._index_shift is not None:
+                null_indices -= profile._index_shift
+                null_indices = [index for index in null_indices if index >= 0]
```

Comment:
```suggestion
                null_indices = [index - profile._index_shift for index in null_indices if index >= 0]
```

Reply:
or something like 

```
shift = profile._index_shift
[index - shift for index in null_indices if index >= 0]
```

Reply:
to condense those two lines into one, the list comprehension would be,
`null_indices = [index - profile._index_shift for index in null_indices if index - profile._index_shift >= 0]`
Could implement it but feels a bit too long for a list comprehension

Reply:
oh fair point.. yeah keep as is